core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -966,6 +971,32 @@ public QueryManagerConfig setFaultTolerantExecutionTaskDescriptorStorageMaxMemor
         return this;
     }
 
+    public DataSize getFaultTolerantExecutionTaskDescriptorStorageHighWaterMark()
+    {
+        return faultTolerantExecutionTaskDescriptorStorageHighWaterMark;
+    }
+
+    @Config("fault-tolerant-execution-task-descriptor-storage-high-water-mark")
+    @ConfigDescription("When memory used to store task descriptors for fault tolerant queries is above high water mark we start to compress the storage")
```

Comment:
Compress the storage when task descriptors in memory are above given data size

Reply:
🏩